src/test/java/de/tracetronic/jenkins/plugins/ecutest/tool/installation/ETConfigurationAsCodeIT.java

```diff
@@ -0,0 +1,98 @@
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import io.jenkins.plugins.casc.model.Sequence;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+/**
+ * {@link ConfigurationAsCode} compatibility tests for {@link ETInstallation}.
+ *
+ * @author Christian Pönisch <christian.poenisch@tracetronic.de>
+ */
+public class ETConfigurationAsCodeIT {
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule jenkins = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void testImportConfiguration() throws Exception {
+        final ETInstallation.DescriptorImpl descriptor = jenkins.jenkins.getDescriptorByType(
+            ETInstallation.DescriptorImpl.class);
+        assertThat(descriptor, notNullValue());
+
+        final ETInstallation[] installations = descriptor.getInstallations();
+        assertThat(installations.length, is(2));
+
+        ETInstallation installation = installations[0];
+        assertThat(installation.getName(), is("ECU-TEST"));
+        assertThat(installation.getHome(), is("C:\\ECU-TEST"));
+        assertThat(installation.getProgId(), is(ETComProperty.DEFAULT_PROG_ID));
+        assertThat(installation.getTimeout(), is(ETComProperty.DEFAULT_TIMEOUT));
+        assertThat(installation.getProperties(), empty());
+
+        installation = installations[1];
+        assertThat(installation.getName(), is("ECU-TEST 7.2"));
+        assertThat(installation.getHome(), is("C:\\Program Files\\ECU-TEST 7.2"));
+        assertThat(installation.getProgId(), is("ECU-TEST.Application.7.2"));
+        assertThat(installation.getTimeout(), is(60));
+        assertThat(installation.getProperties().size(), is(1));
+        assertEquals(installation.getProperties().get(0).getClass(), ETToolProperty.class);
+        assertThat(((ETToolProperty) installation.getProperties().get(0)).getProgId(), is("ECU-TEST.Application.7.2"));
+        assertThat(((ETToolProperty) installation.getProperties().get(0)).getTimeout(), is(60));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void testExportConfiguration() throws Exception {
+        final ETInstallation.DescriptorImpl etDescriptor = jenkins.jenkins
+            .getDescriptorByType(ETInstallation.DescriptorImpl.class);
+        etDescriptor.setInstallations(new ETInstallation("ECU-TEST", "C:\\ECU-TEST",
```

Comment:
Not sure you need to override the configuration, could validate the config loaded in from `@ConfiguredWithCode("configuration-as-code.yml")` :see_no_evil:

Reply:
You're absolutely right! I'll fix it.

Reply:
Great now your fully validating the YAML config :sheep: